languini/train_lib/lm_trainer.py

```diff
@@ -128,14 +123,12 @@ def evaluation(config, model, state, data_source, max_steps, last_n=-1, print_pr
                 check(all_losses, (bsz * last_n,))
 
                 # mask losses that are padded (unlike training, evaluation can result in batches with padded batches)
-                if is_padded:
-                    all_losses = all_losses.masked_fill(batch_x.reshape(-1) == 0, 0.0)
-                    token_count = torch.sum(batch_x != 0)
-                else:
-                    token_count = torch.tensor([batch_x.numel()], device=all_losses.device)
+                is_padding = batch_x.reshape(-1) == 0 # (bsz * last_n,)
```

Comment:
Doing this always now, no matter if `is_padded` or not. Simplifies code and shouldn't lead to any noticeable slowdown. If you prefer, will reintroduce distinction and introduce a special is_padding=None case for `total_correct`